logback-classic/src/main/java/ch/qos/logback/classic/spi/ThrowableProxyUtil.java

```diff
@@ -76,24 +77,35 @@ static int findNumberOfCommonFrames(StackTraceElement[] steArray, StackTraceElem
     public static String asString(IThrowableProxy tp) {
         StringBuilder sb = new StringBuilder(BUILDER_CAPACITY);
 
-        recursiveAppend(sb, null, REGULAR_EXCEPTION_INDENT, tp);
+        if (tp != null) {
+            recursiveAppend(sb, null, null, REGULAR_EXCEPTION_INDENT, tp);
+        }
 
         return sb.toString();
     }
 
-    private static void recursiveAppend(StringBuilder sb, String prefix, int indent, IThrowableProxy tp) {
-        if (tp == null)
-            return;
-        subjoinFirstLine(sb, prefix, indent, tp);
+    private static void recursiveAppend(StringBuilder sb, String prefix, String suffix, int indent, IThrowableProxy tp) {
```

Comment:
Would you prefer I move the map creation to the public constructor?  It would eliminate the lazy construction in the private constructor but would require the creation of the map even if the throwable has no cause or suppressed exceptions.